python/samba/provision/__init__.py

```diff
@@ -37,11 +37,15 @@
 import time
 import uuid
 import socket
-import urllib
 import string
 import tempfile
 import samba.dsdb
 
+try:
+    from urllib.parse import quote  # py3
+except ImportError:
+    from urllib import quote  # fall back to py2
+
```

Comment:
can't we use six here  (which will hide the try/except handling)
e.g.
    from six.moves.urllib.parse import quote
    
 or if not and we want to reinvent then a wrapping function (hiding the try/except) should go in python/samba/compat.py